graylog2-shared/src/main/java/org/graylog2/shared/plugins/PluginLoader.java

```diff
@@ -39,12 +48,17 @@ public PluginLoader(File pluginDir) {
     }
 
     public Set<Plugin> loadPlugins() {
-        final ImmutableSet.Builder<Plugin> plugins = ImmutableSet.builder();
+        final Set<Plugin> sorted = ImmutableSortedSet.orderedBy(new PluginComparator())
+                .addAll(loadClassPathPlugins())
+                .addAll(loadJarPlugins())
+                .build();
```

Comment:
The `Plugin` interface doesn't require `equals` and `hashcode` and if plugin authors don't override those methods, the set could still contain duplicates (i. e. same plugin in different versions) as the default method implementations in `Object` will be used.